### PR TITLE
feature: script for moving item assets to given class

### PR DESCRIPTION
**Pull request by Karol-Stelmaczonek**
Ticket: https://oat-sa.atlassian.net/browse/PISA25-557

## What's Changed

- Create script for moving item assets to given class

## How to test
Run script in console : php index.php 'oat\taoMediaManager\scripts\MoveResource' [--id] [--namePattern] [--destinationClass]
It can move resource by its id or name pattern, you must specify destination class uri.

Demo:

Moving classes: 

https://github.com/oat-sa/extension-tao-mediamanager/assets/118974926/56324f62-c691-4343-a86d-3610962b3b0b

### CODE REVIEW

File: scripts/MoveResources.php [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    protected function run(): Report
+    {
+        $this->report = Report::createInfo(__('Move resource script start.'));
+        $preflightCheck = $this->checkRequiredOptions();
+        if ($preflightCheck === false) {
+            return $this->report;
+        }
+        $destinationClassId = $this->getOption(self::OPTION_DESTINATION_CLASS_URI);
+
+        $mediaInstances = $this->getMediaService()->getRootClass()->getSubClasses(true);
```
Comment (andreluizmachado) [inline scripts/MoveResources.php:90]: What does it mean the argument true on the getSubClasses?

File: scripts/MoveResources.php [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+     * @throws \common_exception_Error
+     */
+    protected function run(): Report
+    {
+        $this->report = Report::createInfo(__('Move resource script start.'));
+        $preflightCheck = $this->checkRequiredOptions();
+        if ($preflightCheck === false) {
+            return $this->report;
+        }
+        $destinationClassId = $this->getOption(self::OPTION_DESTINATION_CLASS_URI);
```
Comment (andreluizmachado) [inline scripts/MoveResources.php:88]: better to check if the class exists before moving it

File: scripts/MoveResources.php (hunk 1 above)
Comment (Karol-Stelmaczonek) [reply scripts/MoveResources.php:90]: Its `getSubClasses($recursive = false)` so I want to recursively get all sub classes but, now I don't know its a good idea. Probably we could stick for just first level, not all classes.
https://github.com/oat-sa/generis/blob/b7b6a3bc27afeab2f5e3c1fbc2ef1a96be5be945/core/kernel/classes/class.Class.php#L70

File: scripts/MoveResources.php (hunk 2 above)
Comment (Karol-Stelmaczonek) [reply scripts/MoveResources.php:88]: Added checking if destination class exists

File: scripts/MoveResources.php (hunk 1 above)
Comment (pnal) [reply scripts/MoveResources.php:90]: for our purpose to clean up root class we do not need recursive for sure

Comment (Karol-Stelmaczonek) [reply scripts/MoveResources.php]: I removed the recursive parameter

**Review by pnal — CHANGES_REQUESTED** [review]
Good job!
During testing it locally I added dry run mode: https://github.com/oat-sa/extension-tao-mediamanager/pull/487
@Karol-Stelmaczonek let me know if I can merge it

**Review by pnal — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful
- [ ] Pull request's target is not `master`
- [ ] Commits are following [conventional commits]
- [ ] Commits messages are meaningful
- [ ] Commits are atomic

**Review by pribi — APPROVED** [review]
LGTM. I see there is no easy way to add DI - so we are still using the legacy version.

File: scripts/MoveResources.php [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+                ResourceTransferCommand::TRANSFER_MODE_MOVE
+            )
+        );
+        $this->report->add(Report::createSuccess(__('Moved resource, id %s', $result->getDestination())));
+    }
+
+
+    private function getMediaService(): MediaService
+    {
+        return $this->getServiceManager()->getContainer()->get(MediaService::class);
```
[the remark this answers is not in the document]
Comment (pnal) [reply scripts/MoveResources.php]: unfortunately we do not support yet DI approach for action class itself, but `MediaService` here is registered in DI container and we getting it using proxy method `->getContainer()`

[the remark this answers is not in the document]
Comment (Karol-Stelmaczonek) [reply scripts/MoveResources.php]: I removed usage of getServiceManager()

File: scripts/MoveResources.php [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+use oat\oatbox\reporting\Report;
+use oat\tao\model\resources\Service\ResourceTransferProxy;
+use oat\taoMediaManager\model\MediaService;
+use oat\tao\model\resources\Command\ResourceTransferCommand;
+use oat\tao\model\resources\Contract\ResourceTransferInterface;
+
+/**
+ * sudo -u www-data php index.php 'oat\taoMediaManager\scripts\MoveResource' [--id] [--namePattern] [--destinationClass]
+ */
+class MoveResources extends ScriptAction
```
[the remark this answers is not in the document]
Comment (Karol-Stelmaczonek) [reply scripts/MoveResources.php:37]: We don't have tests for those script classes and I think adding any would be painfull so I prefer to skip it

**Pull request merged**